src/shader_recompiler/frontend/translate/data_share.cpp

```diff
@@ -110,6 +116,50 @@ void Translator::DS_WRITE(int bit_size, bool is_signed, bool is_pair, const GcnI
     }
 }
 
+void Translator::DS_MAX_U32(const GcnInst& inst) {
+    const IR::U32 addr{GetSrc(inst.src[0])};
+    const IR::U32 data{GetSrc(inst.src[1])};
+    const IR::U32 offset = ir.Imm32(u32(inst.control.ds.offset0));
+    const IR::U32 addr_offset = ir.IAdd(addr, offset);
+
+    const IR::U32 old_value = IR::U32(ir.LoadShared(32, false, addr_offset));
```

Comment:
This needs a special IR opcode (something like SharedAtomic(U/S)Max32) so you can call OpAtomic(U/S)Max in spirv backend. The difference between buffers and this is the pointer type, buffers have spv::StorageClass::StorageBuffer while shared atomics have spv::StorageClass::Workgroup